sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -320,6 +331,67 @@ object ResolveDefaultColumns extends QueryErrorsBase
     coerceDefaultValue(analyzed, dataType, statementType, colName, defaultSQL)
   }
 
+  /**
+   * Analyze EXISTS_DEFAULT value.  This skips some steps of analyze as most of the
+   * analysis has been done before.
+   *
+   * VisibleForTesting
+   */
+  def analyzeExistingDefault(field: StructField,
```

Comment:
This is simpler version of analyze.  We may have an opportunity to simplify further, but for now it seems some part of analysis is still needed to resolve some functions like array().  The problematic code of FinishAnalysis was removed though.